base/sparse/sparsematrix.jl

```diff
@@ -1884,48 +1884,61 @@ function setindex!{Tv,Ti,T<:Real}(A::SparseMatrixCSC{Tv,Ti}, x, I::AbstractVecto
 end
 
 
-
 # Sparse concatenation
 
 function vcat(X::SparseMatrixCSC...)
+    Tv = promote_type(map(x->eltype(x.nzval), X)...)
+    Ti = promote_type(map(x->eltype(x.rowval), X)...)
+
+    vcat(map(x->convert(SparseMatrixCSC{Tv,Ti}, x), X)...)
```

Comment:
I didn't fully understand. Input arrays that are of type `{Tv, Ti}` will anyways be no-op for convert. Worst case, most of them get converted once and then passed to vcat, which stuffs it in. How do we end up converting twice?